<launch>

  # <arg name="server" default="localhost"/>

  <rosparam param="using_keyboard">False</rosparam>

  <param name="agent_num" type="int" value="4" />

  <rosparam param="car_ips">['172.16.0.66', '172.16.0.176', '172.16.0.184', '172.16.0.98']</rosparam>

  <rosparam param="poses_origin">[[1.15,0.62],[1.20,2.48],[3.60,2.46],[3.55,0.04]]</rosparam>

  <param name="eps" type="double" value="0.1" />

  <param name="radius" type="double" value="0.1" />

  <rosparam param="is_reset">False</rosparam>

  <rosparam param="is_finish">False</rosparam>

  <rosparam param="unset_list">[]</rosparam>

  <rosparam param="unfinish_list">[]</rosparam>

  <rosparam param="car_targets">[[3.60,2.46],[3.55,0.04],[1.15,0.62],[1.20,2.48]]</rosparam>

  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      # server: $(arg server)
      server: 172.16.0.7
      port: 3883

      update_frequency: 100.0
      frame_id: world

      # Use the VRPN server's time, or the client's ROS time.
      use_server_time: false
      broadcast_tf: true

      # Must either specify refresh frequency > 0.0, or a list of trackers to create
      refresh_tracker_frequency: 1.0
      #trackers:
      #- FirstTracker
      #- SecondTracker
    </rosparam>
  </node>

  <node pkg="vrpn_keyboard_control" type="vrpn_pose.py" name="vrpn_pose" output="screen">

  </node>

  <node pkg="vrpn_keyboard_control" type="rl_onserver.py" name="rl_onserver" output="screen">

    <param name="LASER_RANGE" type="double" value="1.4" />

    <param name="ROBOT_LENGTH" type="double" value="0.25" />

  </node>

  <node pkg="vrpn_keyboard_control" type="keyboard_control_talker.py" name="keyboard_control_talker" output="screen">
  
  </node>

  <node pkg="vrpn_keyboard_control" type="crash_checker.py" name="crash_checker" output="screen">
  
  </node>

  <node pkg="vrpn_keyboard_control" type="keyboard_control_listener.py" name="keyboard_control_listener" output="screen">
  
  </node>

  <node pkg="vrpn_keyboard_control" type="controller.py" name="controller" output="screen">
  
  </node>

  <node pkg="vrpn_keyboard_control" type="Transer.py" name="Transer" output="screen">
  
  </node>

  <node pkg="vrpn_keyboard_control" type="tcp_server.py" name="tcp_server" output="screen">

    <param name="tcp_ip" value="172.16.0.15" />

    <rosparam param="tcp_port">8888</rosparam>
  
  </node>

</launch>